### ISSUE TITLE
Cross is unable to pull images

### ISSUE THREAD

**Issue by dave-tucker** [cross-rs/cross#765]
### Checklist

- [X] I've looked through the [issues and pull requests](https://github.com/cross-rs/cross/issues?q=) for similar reports

### Describe your issue

I just installed via git, which is now the recommendation since #759 was merged

```
❯ cross build --target aarch64-unknown-linux-gnu
Unable to find image 'ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main' locally
docker: Error response from daemon: Head "https://ghcr.io/v2/cross-rs/aarch64-unknown-linux-gnu/manifests/main": denied: denied.
See 'docker run --help'.
```

And also:

```
❯ docker pull ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main
Error response from daemon: Head "https://ghcr.io/v2/cross-rs/aarch64-unknown-linux-gnu/manifests/main": denied: denied
```

I was under the impression that GHCR allowed unauthenticated access for public images... perhaps that has changed, or perhaps the README should document the steps required.

### What target(s) are you cross-compiling for?

aarch64-unknown-linux-gnu

### Which operating system is the host (e.g computer cross is on) running?

- [ ] macOS
- [ ] Windows
- [X] Linux / BSD
- [ ] other OS (specify in description)

### What architecture is the host?

- [X] x86_64 / AMD64
- [ ] arm32
- [ ] arm64 (including Mac M1)

### What container engine is cross using?

- [X] docker
- [ ] podman
- [ ] other container engine (specify in description)

### cross version

cross 0.2.1 (42cea7d 2022-06-07)

### Example

Attempt to build any project, where Docker is installed AND you are not authenticated with GHCR

```sh
 cross build --target aarch64-unknown-linux-gnu
 ```

**Comment by Alexhuszagh** [comment]
> I was under the impression that GHCR allowed unauthenticated access for public images... perhaps that has changed, or perhaps the README should document the steps required.

I believe that's true, I'll add some info but I hope this just a temporary network bug.

**Comment by Alexhuszagh** [comment]
For example, on Windows I get:

```bash
$ docker login ghcr.io
Username:
```

So I'm not logged into any registry. Just for good measure, I haven't even started my SSH agent.

```bash
$ ssh-add -l
Could not open a connection to your authentication agent.
```

I'll get a list of images, and then fetch one that's unrelated:

```bash
$ docker images
REPOSITORY                                   TAG       IMAGE ID       CREATED      SIZE
ghcr.io/cross-rs/aarch64-linux-android       main      7ec45f56928b   4 days ago   3.99GB
ghcr.io/cross-rs/armv7-linux-androideabi     main      1d54871b6522   4 days ago   3.98GB
ghcr.io/cross-rs/aarch64-unknown-linux-gnu   main      8602d2945566   4 days ago   1.28GB
$ docker pull ghcr.io/cross-rs/armv7-unknown-linux-gnueabihf:main
main: Pulling from cross-rs/armv7-unknown-linux-gnueabihf
58690f9b18fc: Already exists
b51569e7c507: Already exists
da8ef40b9eca: Already exists
fb15d46c38dc: Already exists
c4bb01d5bb74: Pull complete
ca5dd73b2a10: Extracting [==========================================>        ]  154.9MB/180.4MB
924ddcdc32dc: Download complete
1d41e58f9108: Download complete
c2be8d2d6b2d: Download complete
ae09f147eae0: Download complete
e970f4bd429e: Download complete
e6b3d772b63a: Download complete
9afb644bef2b: Download complete
681a2c8230c9: Download complete
fa5d991fb606: Download complete
982b8e466b81: Download complete
f342bbfeaad7: Download complete
970d552e4c73: Download complete
```

I get the same for WSL2, Linux native (Fedora 36), and macOS Catalina.

**Comment by Emilgardis** [comment]
Are you behind somekind of company proxy maybe?

**Comment by dave-tucker** [comment]
> Are you behind somekind of company proxy maybe?

I am not.

@Alexhuszagh aha! so I did a quick `cat ~/.docker/config.json` and it turns out I was authenticated - must have been when playing with GHCR a while back. Removing the auth for ghcr did the trick... I don't recall which token I used, but it would be nice to be allowed access to other public repos while authenticated to access private ones... but that's not a cross issue.

Thanks for the help!
